src/base/bittorrent/session.cpp

```diff
@@ -3691,12 +3663,22 @@ void Session::handleTorrentMetadataReceived(TorrentHandle *const torrent)
     saveTorrentResumeData(torrent);
 
     // Save metadata
-    const QDir resumeDataDir(m_resumeFolderPath);
-    QString torrentFile = resumeDataDir.absoluteFilePath(QString("%1.torrent").arg(torrent->hash()));
-    if (torrent->saveTorrentFile(torrentFile)) {
-        // Copy the torrent file to the export folder
-        if (!torrentExportDirectory().isEmpty())
-            exportTorrentFile(torrent);
+    QByteArray metadata{torrent->metadata()};
+    if (!metadata.isEmpty()) {
+        using namespace DB;
+        QSqlDatabase db = QSqlDatabase::database(MAIN_CONNECTION_NAME);
+        QSqlQuery query(db);
+        query.prepare(QString("UPDATE %1 "
+                              "SET %2 =  :metadata "
+                              "WHERE %3 = :hash")
+                      .arg(TABLE_NAME, COL_METADATA, COL_HASH));
```

Comment:
my other recommendation would be to encapsulate this behind some class that just exposes the operations you want to perform.  u.e. "updateHash".  That way you can change things behind the scenes, without having to update all these places.  It also isolates you from leaking out the details of tehse tables/columns and whatnot to all this consumptive code.